cpp/CMakeLists.txt

```diff
@@ -296,6 +318,16 @@ find_path(
     HINTS "$ENV{DLPACK_ROOT}/include"
 )
 
+if(NOT DLPACK_INCLUDE)
+  CPMFindPackage(
+    NAME DLPACK
+    GITHUB_REPOSITORY dmlc/dlpack
+    VERSION 0.3
+  )
```

Comment:
Similar to RMM, users can override with `DLPACK_ROOT` for an isntalled DLPack or `CPM_DLPACK_SOURCE` for a source DLPack.